matrix-sdk/src/main/java/org/matrix/androidsdk/data/timeline/MXEventTimeline.java

```diff
@@ -482,8 +483,10 @@ private void addPaginationEvents(List<Event> events,
             boolean processedEvent = true;
 
             if (event.stateKey != null) {
+                boolean considerNewContent = direction == Direction.FORWARDS;
+
                 deepCopyState(direction);
-                processedEvent = processStateEvent(event, direction);
+                processedEvent = processStateEvent(event, direction, considerNewContent);
```

Comment:
Why don't you "enrich the timeline root state with the additional state events observed during back pagination" here?

Reply:
N/A